### DISCUSSION TITLE
How can I add multiple listener functions for one event? (FIXED)

### DISCUSSION THREAD

**Discussion by Dwyte** [socketio/socket.io#3746] [Q&A] [answered]
Is it possible? I am using socket.io on a react project and I'm trying to listen to one event in 2 different components. The problem is only one of them is working, I think the other one is being replaced. Thanks!

Edit: I found out somewhere in my component tree, all listeners on that event gets turned off. (FIXED)

**Comment by darrachequesne** [comment, marked as answer]
Thanks for the update!

For future readers, you can register any number of listeners for a given event, they will all be called.

See also:

- https://socket.io/docs/v3/listening-to-events/
- https://nodejs.org/api/events.html (used on the server-side)
- https://github.com/component/emitter/ (used on the client-side)
